src/flint/padic.jl

```diff
@@ -738,14 +765,81 @@ end
 #
 ###############################################################################
 
-# inner constructor is also used directly
+# Kept for backwards compatibility; the user facing constructor is `padic_field`
+function PadicField(p::Integer, prec::Int = 64; kw...)
+   return PadicField(ZZRingElem(p), prec; kw...)
+end
 
 @doc raw"""
-    PadicField(p::Integer, prec::Int; kw...)
+    padic_field(p::Integer; precision::Int=64, cached::Bool=true, check::Bool=true)
+    padic_field(p::ZZRingElem; precision::Int=64, cached::Bool=true, check::Bool=true)
 
-Returns the parent object for the $p$-adic field for given prime $p$, where
-the default absolute precision of elements of the field is given by `prec`.
+Return the $p$-adic field for the given prime $p$.
+The default absolute precision of elements of the field may be set with `precision`.
 """
-function PadicField(p::Integer, prec::Int; kw...)
-   return PadicField(ZZRingElem(p), prec; kw...)
+padic_field
+
+function padic_field(p::Integer; precision::Int=64, cached::Bool=true, check::Bool=true)
+  return padic_field(ZZRingElem(p), precision=precision, cached=cached, check=check)
+end
+
+function padic_field(p::ZZRingElem; precision::Int=64, cached::Bool=true, check::Bool=true)
+  return PadicField(p, precision, cached=cached, check=check)
 end
+
+###############################################################################
+#
+#   Precision handling
+#
+###############################################################################
+
+Base.precision(Q::PadicField) = Q.prec_max
+
+function Base.setprecision(q::PadicFieldElem, N::Int)
+  r = parent(q)()
+  r.N = N
+  ccall((:padic_set, libflint), Nothing, (Ref{PadicFieldElem}, Ref{PadicFieldElem}, Ref{PadicField}), r, q, parent(q))
+  return r
+end
+
+function setprecision!(a::PadicFieldElem, n::Int)
+  a.N = n
+  ccall((:padic_reduce, libflint), Nothing, (Ref{PadicFieldElem}, Ref{PadicField}), a, parent(a))
+  return a
+end
+
+function setprecision!(Q::PadicField, n::Int)
```

Comment:
Should we rename this function to something like `_setprecision_internal!` and deprecate it?

Reply:
Hm, let's keep it for now.